### DISCUSSION TITLE
[question] does `cloudnative-pg` allow upgrade major version of PostgreSQL via change version in CRD?

### DISCUSSION THREAD

**Discussion by Slach** [cloudnative-pg/cloudnative-pg#389] [Q&A] [answered]
I search solution which allow upgrade major versions of PostgreSQL in Kubernetes
Could `cloudnative-pg` helps me with this use case?

**Comment by gbartolini** [comment, marked as answer]
We don't allow upgrading the major version in the image name.

Version 1.16.0 introduces support for offline major upgrades through the creation of new clusters.

Check it out: https://cloudnative-pg.io/documentation/1.16/database_import/

We don't support at the moment in place upgrades, and at the moment we do not plan for it as our main recommendation is to go towards microservice databases owned by application developers. Migrating to a new major version of PostgreSQL might require going through application testing that should be done in a different cluster.

**Reply by StuartAAustin** [reply]
Using CNPG 22.1 now and wondered if there is any change to support major version upgrades of postgresql.

**Reply by gbartolini** [reply]
Please check this update: https://www.gabrielebartolini.it/articles/2024/03/cloudnativepg-recipe-5-how-to-migrate-your-postgresql-database-in-kubernetes-with-~0-downtime-from-anywhere/
